src/main/flight/mixer.c

```diff
@@ -346,14 +346,92 @@ static void applyFlipOverAfterCrashModeToMotors(void)
     }
 }
 
+#ifdef USE_RPM_LIMITER
+static void applyRPMLimiter(void)
+{
+    if (mixerConfig()->rpm_limiter && motorConfig()->dev.useDshotTelemetry) {
+        float scaledRPMLimit = 0;
+        float pidOutput = 0;
+        float rcCommandThrottle = (rcCommand[THROTTLE] - 1000) / 1000.0f;
+        float averageRPM = 0;
+        float averageRPMSmoothed = 0;
+        bool motorsSaturated = false;
+        for (int i = 0; i < getMotorCount(); i++) {
+            averageRPM += getDshotTelemetry(i);
+            if (motor[i] >= motorConfig()->maxthrottle) {
+                motorsSaturated = true;
+            }
+        }
+        averageRPM = 100 * averageRPM / (getMotorCount() * motorConfig()->motorPoleCount / 2.0f);
```

Comment:
This line of code is a good reminder that we should just have one place in the code where RPM is calculated, and then just pass it around to all the other parts of code that need it.

Reply:
Would this also be the opportunity to read that value is as a UINT16 so we can end-run around the identified overflow at 32767, and sanity check that behavior?